psycopg/psycopg/types/composite.py

```diff
@@ -216,64 +201,95 @@ def __init__(self, oid: int, context: abc.AdaptContext | None = None):
         self._txs: dict[tuple[int, ...], abc.Transformer] = {}
 
     def load(self, data: abc.Buffer) -> tuple[Any, ...]:
-        nfields = unpack_len(data, 0)[0]
-        offset = 4
-        oids = []
-        record: list[Buffer | None] = []
-        for _ in range(nfields):
-            oid, length = _unpack_oidlen(data, offset)
-            offset += 8
-            oids.append(oid)
-            if length >= 0:
-                record.append(data[offset : offset + length])
-                offset += length
-            else:
-                record.append(None)
-
-        key = tuple(oids)
-        try:
-            tx = self._txs[key]
-        except KeyError:
-            tx = self._txs[key] = Transformer(self._ctx)
-            tx.set_loader_types(oids, self.format)
+        record, oids = _parse_binary_record(data)
+        if not record:
+            return ()
 
+        tx = self._get_transformer(tuple(oids))
         return tx.load_sequence(record)
 
+    def _get_transformer(self, key: tuple[int, ...]) -> abc.Transformer:
+        # This could be managed with a custom `dict.__missing__` implementation
+        # micro-optimization - Prefer a membership check to a KeyError
+        #   try/except blocks are expensive in hot loops.
```

Comment:
This is not necessarily true. It used to be, but [since Python 3.11](https://bugs.python.org/issue40222) you pay the cost only if the exception is captured, not when it is set up.

I don't mind the refactoring, let's keep it, but the comment is redundant.

```suggestion
```